packages/memory-inspector/src/browser/memory-widget/memory-options-widget.tsx

```diff
@@ -0,0 +1,726 @@
+            this.errorTimeout = undefined;
+            return;
+        }
+        this.showMemoryError = true;
+        this.update();
+        this.errorTimeout = setTimeout(() => {
+            this.showMemoryError = false;
+            this.update();
+            this.errorTimeout = undefined;
+        }, Constants.ERROR_TIMEOUT);
+    };
+
+    protected doRefresh = (event: React.KeyboardEvent<HTMLInputElement> | React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+        if ('key' in event && event.key !== 'Enter') {
+            return;
+        }
+        this.updateMemoryView();
+    };
+
+    // eslint-disable-next-line @typescript-eslint/member-ordering
+    protected updateMemoryView = debounce(this.doUpdateMemoryView.bind(this), Constants.DEBOUNCE_TIME, { trailing: true });
+
+    protected async doUpdateMemoryView(): Promise<void> {
+        if (!(this.addressField && this.readLengthField)) { return; }
+
+        if (this.addressField?.value.trim().length === 0) {
+            this.memoryReadError = nls.localize('theia/memory-inspector/memory/addressField/memoryReadError', 'Enter an address or expression in the Location field.');
+            this.doShowMemoryErrors();
+            return;
+        }
+        if (this.readLengthField.value.trim().length === 0) {
+            this.memoryReadError = nls.localize('theia/memory-inspector/memory/readLength/memoryReadError', 'Enter a length (decimal or hexadecimal number) in the Length field.');
+            this.doShowMemoryErrors();
+            return;
+        }
+
+        const startAddress = this.addressField.value;
+        const locationOffset = parseInt(`${this.offsetField?.value}`) || 0;
+        const readLength = parseInt(this.readLengthField.value);
+
+        try {
+            this.memoryReadResult = await this.getMemory(startAddress, readLength, locationOffset);
+            this.fireDidChangeMemory();
+            if (this.pinnedMemoryReadResult) {
+                this.pinnedMemoryReadResult.resolve(this.memoryReadResult);
+            }
+            this.doShowMemoryErrors(true);
+        } catch (err) {
+            this.memoryReadError = this.getUserError(err);
+            console.error('Failed to read memory', err);
+            this.doShowMemoryErrors();
+            if (this.pinnedMemoryReadResult) {
+                this.pinnedMemoryReadResult.resolve(this.memoryReadResult);
+            }
+        } finally {
+            this.pinnedMemoryReadResult = undefined;
+            this.update();
+        }
+    }
+
+    protected getUserError(err: unknown): string {
+        return err instanceof Error ? err.message : nls.localize('theia/memory-inspector/memory/userError', 'There was an error fetching memory.');
+    }
+
+    protected async getMemory(memoryReference: string, count: number, offset: number): Promise<Interfaces.MemoryReadResult> {
+        const result = await this.memoryProvider.readMemory({ memoryReference, count, offset });
```

Comment:
In our customization we had to extend the read memory arguments so it would have made things a bit easier if that call were in a separate protected method.

Reply:
Done.